01-import_and_maxfilter.py

```diff
@@ -137,8 +137,45 @@ def find_bad_channels(raw, subject, session):
     logger.info(gen_log_message(message=msg, step=1,
                                 subject=subject, session=session))
 
+    # Write the bad channels to disk.
+    deriv_path = config.get_subject_deriv_path(subject=subject,
+                                               session=session,
+                                               kind=config.get_kind())
+
+    bids_basename = make_bids_basename(subject=subject,
+                                       session=session,
+                                       task=config.get_task(),
+                                       acquisition=config.acq,
+                                       run=run,
+                                       processing=config.proc,
+                                       recording=config.rec,
+                                       space=config.space)
+
+    bads_tsv_fname = op.join(deriv_path, f'{bids_basename}_bad_chans.tsv')
+    bads_for_tsv = []
+    reasons = []
+
+    if config.find_flat_channels_meg:
+        bads_for_tsv.extend(auto_flat_chs)
+        reasons.extend(['auto-flat'] * len(auto_flat_chs))
+        preexisting_bads = set(preexisting_bads) - set(auto_flat_chs)
+
+    if config.find_noisy_channels_meg:
+        bads_for_tsv.extend(auto_noisy_chs)
+        reasons.extend(['auto-noisy'] * len(auto_noisy_chs))
+        preexisting_bads = set(preexisting_bads) - set(auto_noisy_chs)
+
+    preexisting_bads = list(preexisting_bads)
+    if preexisting_bads:
+        bads_for_tsv.extend(preexisting_bads)
+        reasons.extend(['pre-existing (before mne-study-template was run)'] *
+                       len(preexisting_bads))
+
+    tsv_data = pd.DataFrame(dict(Channel=bads_for_tsv, Reason=reasons))
```

Comment:
```suggestion
    tsv_data = pd.DataFrame(dict(name=bads_for_tsv, reason=reasons))
```
to follow the convention of _channels.tsv files.